### PR TITLE
arm/src/up_doirq.c: move local variable definition into proper conditional

### PR DESCRIPTION
**Pull request by abhishek-samsung**
Fix the following build warnings:

CC: armv8-m/up_doirq.c
armv8-m/up_doirq.c: In function 'up_doirq':
armv8-m/up_doirq.c:95:11: warning: unused variable 'stack_remain' [-Wunused-variable]
uint32_t stack_remain;
^~~~~~~~~~~~
CC: armv7-m/up_doirq.c
armv7-m/up_doirq.c: In function 'up_doirq':
armv7-m/up_doirq.c:95:11: warning: unused variable 'stack_remain' [-Wunused-variable]
uint32_t stack_remain;
^~~~~~~~~~~~

### CODE REVIEW

File: os/arch/arm/src/armv7-m/up_doirq.c [hunk 1]
```diff
@@ -104,6 +103,7 @@ uint32_t *up_doirq(int irq, uint32_t *regs)
 #ifdef CONFIG_ARCH_NESTED_INTERRUPT_STACKCHECK
 
 	/* check remaining stack available, assert if stack overflows */
+	uint32_t stack_remain;
```
Comment (sangam-swami) [inline os/arch/arm/src/armv7-m/up_doirq.c:106]: PR description says error message but it looks warning. can you change description

Comment (sunghan-chang) [reply os/arch/arm/src/armv7-m/up_doirq.c:106]: And how about titling like `arm/src/armv7-m/up_doirq.c: remove unused variable`? This is better to know what it is I think.

Comment (abhishek-samsung) [reply os/arch/arm/src/armv7-m/up_doirq.c:106]: Done

**Review by jeongchanKim — DISMISSED** [review]
LGTM

**Comment by sunghan-chang** [comment]
@abhishek-samsung Oh really sorry. I left wrong suggestion for the title. I knew it was removal. But it's moving into proper conditional. Could you rewrite the title?
i.e. `.... : move local variable definition into proper conditional`

**Review by sangam-swami — APPROVED** [review]
looks good

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**